### PR TITLE
Change cuda image for tensorRT to the one with cudnn8

### PR DESCRIPTION
**Pull request by jchen351**
### Description
copilot:summary

### Motivation and Context
copliot::walkthrough

### CODE REVIEW

**Comment by jchen351** [comment]
copilot:summary

**Comment by jchen351** [comment]
Fixing issue from #18000

File: tools/ci_build/github/azure-pipelines/templates/linux-gpu-tensorrt-packaging-pipeline.yml [hunk 1]
```diff
@@ -48,7 +48,7 @@ stages:
           Context: tools/ci_build/github/linux/docker
           DockerBuildArgs: "
           --network=host
-          --build-arg BASEIMAGE=nvidia/cuda:11.8.0-devel-ubi8
+          --build-arg BASEIMAGE=nvidia/cuda:11.8.0-cudnn8-devel-ubi8
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/templates/linux-gpu-tensorrt-packaging-pipeline.yml:51]: There is no such an image for CUDA 12.  Have you tried to build an image for CUDA 12?

From what I see the following lines in the docker file seem not working:

```
CUDA_VERSION=$(nvcc --version | sed -n 's/^.*release \([0-9]\+\.[0-9]\+\).*$/\1/p') &&  \
     dnf -y install  \
     libcudnn8-devel-*cuda${CUDA_VERSION}*  \
     libcudnn8-*cuda${CUDA_VERSION}* ; \
```

Comment (jchen351) [reply tools/ci_build/github/azure-pipelines/templates/linux-gpu-tensorrt-packaging-pipeline.yml:51]: nvidia/cuda:12.2.2-cudnn8-devel-ubi8

**Review by snnn — APPROVED** [review]

**Pull request merged**